make/test.mk

```diff
@@ -76,6 +76,22 @@ e2e: $(BINDIR)/scratch/kind-exists | $(NEEDS_KUBECTL) $(NEEDS_GINKGO)
 e2e-ci: e2e-setup-kind e2e-setup
 	make/e2e-ci.sh
 
+$(BINDIR)/test/e2e.test: FORCE | $(NEEDS_GINKGO) $(BINDIR)/test
+	$(GINKGO) build --tags e2e_test test/e2e
+	mv test/e2e/e2e.test $(BINDIR)/test/e2e.test
+
+.PHONY: e2e-build
```

Comment:
Some more information I would like to see as a user who wants to use this target:
- that it will talk to cert-manager installation in cert-manager namespace in cluster pointed at by current kubecontext (if that is the case) and will create resources in that cluster
- some kind of warning that some/most tests will require some dependencies (ingress) so I need to pick which tests I actually want to run using ginkgo.skip/ginkgo.focus
- that tests are tied to the specific cert-manager version (can this test be run against cert-manager v1.11)?
- an example how to use the built binary (like the one you posted in issue description `_bin/test/e2e.test --repo-root=/dev/null --ginkgo.focus="CA\ Issuer" --ginkgo.skip="Gateway"`)

Reply:
Done. See what you think.

Reply:
Thanks for making the changes Richard, this looks great to me 👍🏼